src/main/main.c

```diff
@@ -591,6 +613,58 @@ static int poll_setup(struct re *re)
 }
 
 
+static void fhs_destroy(void *data)
+{
+	struct fhs *fhs = data;
+
+	hash_unlink(&fhs->he);
+}
+
+
+static int fhs_update(struct re *re, struct fhs **fhsp, re_sock_t fd,
+		      int flags, fd_h *fh, void *arg)
+{
+	struct fhs *fhs = NULL;
+	struct le *he = hash_lookup(re->fhl,
+				    hash_fast_murmur((uint32_t)fd, re->maxfds),
```

Comment:
On Windows `re_sock_t` is likely 64-bit `size_t`, so you probably want to mix the higher half as well. May I suggest to create a `hash_sock_t` function that will do that on Windows and forward the result to `hash_fast_murmur`. On POSIX it would call `hash_fast_murmur` straight away.